include/jemalloc/internal/atomic_c11.h

```diff
@@ -0,0 +1,100 @@
+#ifndef JEMALLOC_INTERNAL_ATOMIC_C11_H
+#define JEMALLOC_INTERNAL_ATOMIC_C11_H
+
+#include <stdatomic.h>
+
+#define ATOMIC_INIT(...) ATOMIC_VAR_INIT(__VA_ARGS__)
+
+#define atomic_memory_order_t memory_order
+#define atomic_memory_order_relaxed memory_order_relaxed
+#define atomic_memory_order_acquire memory_order_acquire
+#define atomic_memory_order_release memory_order_release
+#define atomic_memory_order_acq_rel memory_order_acq_rel
+#define atomic_memory_order_seq_cst memory_order_seq_cst
+
+#define atomic_fence atomic_thread_fence
+
+#define JEMALLOC_GENERATE_ATOMICS(type, short_type, /* unused */ lg_size)      \
+typedef _Atomic(type) atomic_##short_type##_t;                                 \
+                                                                               \
+ATOMIC_INLINE type                                                             \
+atomic_load_##short_type(const atomic_##short_type##_t *a,                     \
+    atomic_memory_order_t mo) {                                                \
+	/*                                                                     \
+	 * A strict interpretation of the C standard prevents atomic_load from \
+	 * taking a const argument, but it's convenient for our purposes.      \
+	 * This cast is a workaround.                                          \
+	 */                                                                    \
+	atomic_##short_type##_t* a_nonconst = (atomic_##short_type##_t*)a;     \
```

Comment:
Does this give the compiler the option of hoisting calls to this function, such that the atomic load isn't guaranteed to happen every time this function is "called"?